### PR TITLE
Hash tx metadata before signing

### PR DESCRIPTION
**Pull request by rvl**
### Issue Number

ADP-800

### Overview

When signing tx metadata using the `signMetadata` endpoint, hash the message first before signing.

### CODE REVIEW

File: lib/core-integration/src/Test/Integration/Scenario/API/Shelley/Wallets.hs [hunk 1]
```diff
@@ -1115,9 +1115,10 @@ spec = describe "SHELLEY_WALLETS" $ do
         let dummyChainCode = BS.replicate 32 0
         let sig = CC.xsignature $ BL.toStrict $ getFromResponse id rSig
         let key = unsafeXPub $ fst (getFromResponse #getApiVerificationKey rKey) <> dummyChainCode
-        let msg = unsafeFromHex "A10071706C65617365207369676E20746869732E"
+        let _msg = unsafeFromHex "A10071706C65617365207369676E20746869732E"
```
Comment (paweljakubas) [inline lib/core-integration/src/Test/Integration/Scenario/API/Shelley/Wallets.hs]: this one probably we want to delete, right?

File: lib/core/src/Cardano/Wallet.hs [hunk 2]
```diff
@@ -2001,8 +2003,11 @@ signMetadataWith ctx wid pwd (role_, ix) metadata = db & \DBLayer{..} -> do
             let DerivationPrefix (_, _, acctIx) = derivationPrefix (getState cp)
             let acctK = deriveAccountPrivateKey encPwd rootK acctIx
             let addrK = deriveAddressPrivateKey encPwd acctK role_ addrIx
-            let msg   = serialiseToCBOR metadata
-            pure $ Signature $ BA.convert $ CC.sign encPwd (getRawKey addrK) msg
+            pure $
+                Signature $ BA.convert $
+                CC.sign encPwd (getRawKey addrK) $
+                hash @ByteString @Blake2b_256 $
```
Comment (paweljakubas) [inline lib/core/src/Cardano/Wallet.hs:2009]: :+1:

**Review by paweljakubas — APPROVED** [review]
lgtm! wonder if we can add some golden testing basing on some already existent test in cardano-node?

File: lib/core-integration/src/Test/Integration/Scenario/API/Shelley/Wallets.hs (hunk 1 above)
Comment (rvl) [reply lib/core-integration/src/Test/Integration/Scenario/API/Shelley/Wallets.hs]: Yeah, the string is already shown in the comment above. I deleted it.

**Comment by rvl** [comment]
I added a cheap "golden" signature check to the same integration test.

**Pull request merged**